### PR TITLE
Allow to specify CLI options when calling check_url

### PR DESCRIPTION
**Pull request by mlocher**

### CODE REVIEW

**Comment by mikeatlas** [comment]
Hmm I encourage you to reconsider... skipping SSL cert verification should be a deliberate opt-out action with good rationale behind it. Default behavior _should_ fail if SSL cert chain fails. Otherwise, by default, `check_url.sh` is prone to potential MitM attacks.

**Comment by mikeatlas** [comment]
For example, some reading material.

https://blog.cloudflare.com/introducing-strict-ssl-protecting-against-a-man-in-the-middle-attack-on-origin-traffic/
https://news.ycombinator.com/item?id=8555026

File: utilities/check_url.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# Ensure a separate script is called in the event of an error
+#
+# Include in your builds via
+# \curl -sSL https://raw.githubusercontent.com/codeship/scripts/master/utilties/check_url.sh > ${HOME}/bin/check_url && chmod u+x ${HOME}/bin/check_url
+#
+# then use the script in your tests like
+# check_url -t 6 -w 10 -o "--no-check-certificate --output-document=/dev/null" "url_to_check"
+OPTIONS="--no-check-certificate --output-document=/dev/null"
```
Comment (mikeatlas) [inline utilities/check_url.sh]: `--no-check-certificate` should be _opt-in_ and only for really good reasons...

Some reading material to convince you:
- http://mislav.uniqpath.com/2013/07/ruby-openssl/
- https://blog.cloudflare.com/introducing-strict-ssl-protecting-against-a-man-in-the-middle-attack-on-origin-traffic/
- https://news.ycombinator.com/item?id=8555026

**Comment by mlocher** [comment]
@mikeatlas I agree, I removed the option from the default options (and made some other improvements as well).

For the version included on Codeship we will leave the option on by default as to not break existing builds though.

**Comment by mikeatlas** [comment]
Hmm, okay... Do you ever consider acting like heroku and doing an audit on customers affected by the `--no-check-certificate` option? For example, you may have only 2% of your userbase calling `check_url` against invalid or self-signed certificate URLs; notifying them of the change might be a lot better than leaving it in place for backwards compatibility (and backwards insecurity).

**Comment by mlocher** [comment]
I'll think about a way to print a warning message if the first check fails because of an invalid certificate. Note that you can always override the included version via the command shown in https://github.com/codeship/scripts/pull/48/files#diff-29b90d12f6f05f0846d37cde4dedbe95R5

**Comment by mikeatlas** [comment]
Thanks, yes, for the record I'm more adamant about the right way to do this in general over our specific suite of tests being "safe" as a Codeship customer.

There's WAY too much incorrect information on how to "overcome SSL validation failed" solutions on Q/A sites like StackOverflow and other blog posts which suggest "just use the `NO_VALIDATE_PEER` option" as the "acceptable workaround", which inevitably ends up being in codebases deployed to production systems.

I certainly hope this PR itself gains SEO traction on the topic of why `--no-check-certificate` is a bad idea that you should really think twice about in using `wget` within your scripting suites/environments.

**Pull request merged**